reference.py

```diff
@@ -61,9 +61,21 @@ def get_pubkey_from_input(vin: VinInfo) -> ECPubKey:
         if (pubkey.valid) & (pubkey.compressed):
             return pubkey
     if is_p2tr(vin.prevout):
+        witnessStack = vin.txinwitness.scriptWitness.stack
+        if (len(witnessStack) > 1):
+            # Script spend
+            control_block = witnessStack[-1]
+            #  control block is <control byte> <32 byte internal key> or more <32 byte hash>
+            internal_key = control_block[1:33].hex()
+            if (internal_key == NUMS_POINT[0].to_bytes(32, 'big').hex()):
```

Comment:
seems a bit odd to convert the internal key to hex and also convert the nums point to hex, cant we just compare the byte arrays?

Reply:
see https://github.com/S3RK/bip-352/pull/3#discussion_r1455539907

Reply:
Resolved in https://github.com/S3RK/bip-352/pull/3/commits/24a64917fcc66cdaa4ba18273468e9a0333d399f